### PR TITLE
Avoid infinite recursion in __torch_function__ example

### LINKED ISSUE #55284: __torch_function__ documentation for inheriting from tensor seems incomplete or incorrect
In the `__torch_function__` [documentation](https://pytorch.org/docs/stable/notes/extending.html) an example class is provided:

```
class LoggingTensor(torch.Tensor):
    @classmethod
    def __torch_function__(cls, func, types, args=(), kwargs=None):
        logging.info(f"func: {func.__name__}, args: {args!r}, kwargs: {kwargs!r}")
        if kwargs is None:
            kwargs = {}
        return super().__torch_function__(func, types, args, kwargs)
```

On nightly if the following program is run:

```
import torch
import logging

class LoggingTensor(torch.Tensor):
    @classmethod
    def __torch_function__(cls, func, types, args=(), kwargs=None):
        logging.info(f"func: {func.__name__}, args: {args!r}, kwargs: {kwargs!r}")
        if kwargs is None:
            kwargs = {}
        return super().__torch_function__(func, types, args, kwargs)

t = LoggingTensor((0.,))
print(t)
```

Then the print will produce

```
RecursionError: maximum recursion depth exceeded
```

cc @hameerabbasi @rgommers @peterbell10 @brianjo @mruberry

### PR DESCRIPTION
**Pull request by peterbell10**
Fixes #55284

This gets the example to run but probably doesn't help the readability of the example. 

Thoughts?

### CODE REVIEW

**Review by hameerabbasi — APPROVED** [review]
LGTM, thanks. I hate special casing so I can see why I didn't consider this fix right away.

**Comment by mruberry** [comment]
Thanks @peterbell10! This fix makes sense to me but I've requested a review from @ezyang.

File: docs/source/notes/extending.rst [hunk 1]
```diff
@@ -517,7 +517,9 @@ calls::
   class LoggingTensor(torch.Tensor):
       @classmethod
       def __torch_function__(cls, func, types, args=(), kwargs=None):
-          logging.info(f"func: {func.__name__}, args: {args!r}, kwargs: {kwargs!r}")
+          # NOTE: Logging a tensor calls Tensor.__repr__, so we can't log __repr__ without infinite recursion
+          if func != torch.Tensor.__repr__:
```
Comment (hameerabbasi) [inline docs/source/notes/extending.rst:521]:
```suggestion
          if func is not torch.Tensor.__repr__:
```

**Review by hameerabbasi — COMMENTED** [review]
Perhaps a small modification.

**Comment by ezyang** [comment]
just goes to show we need to run our doc examples haha

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**